events/api.py

```diff
@@ -718,30 +726,52 @@ class EventSerializer(LinkedEventsSerializer, GeoModelAPIView):
     publication_status = EnumChoiceField(PUBLICATION_STATUSES)
     external_links = EventLinkSerializer(many=True, required=False)
     offers = OfferSerializer(many=True, required=False)
+    id = serializers.CharField(required=False)
+    data_source = serializers.PrimaryKeyRelatedField(queryset=DataSource.objects.all(),
+                                                     required=False)
+    publisher = serializers.PrimaryKeyRelatedField(queryset=Organization.objects.all(),
+                                                   required=False)
     sub_events = JSONLDRelatedField(serializer='EventSerializer',
                                     required=False, view_name='event-detail',
                                     many=True, queryset=Event.objects.all())
-    id = serializers.ReadOnlyField()
-    data_source = serializers.PrimaryKeyRelatedField(read_only=True)
-    publisher = serializers.PrimaryKeyRelatedField(read_only=True)
     image = JSONLDRelatedField(serializer=ImageSerializer, required=False, allow_null=True,
-                                     view_name='image-detail', queryset=Image.objects.all(), expanded=True)
+                               view_name='image-detail', queryset=Image.objects.all(), expanded=True)
     in_language = JSONLDRelatedField(serializer=LanguageSerializer, required=False,
                                      view_name='language-detail', many=True, queryset=Language.objects.all())
     audience = JSONLDRelatedField(serializer=KeywordSerializer, view_name='keyword-detail',
                                   many=True, required=False, queryset=Keyword.objects.all())
     view_name = 'event-detail'
     # JA: TODO: Temporarily disabled new validation code
-    fields_needed_to_publish = ('keywords', 'location', 'start_time') #,
-                                # 'short_description', 'description',
-                                # 'offers')
+    fields_needed_to_publish = ('keywords', 'location', 'start_time')  # ,
+    # 'short_description', 'description',
+    # 'offers')
 
     def __init__(self, *args, skip_empties=False, **kwargs):
         super(EventSerializer, self).__init__(*args, **kwargs)
         # The following can be used when serializing when
         # testing and debugging.
         self.skip_empties = skip_empties
 
+        # for post and put methods, user information is needed to restrict permissions at validate
+        self.method = self.context['request'].method
+        if self.context['request'].method in ['PUT', 'POST']:
+            self.user = self.context['request'].user
+            self.api_key = self.context['request'].query_params.get('api_key')
```

Comment:
This doesn't seem like the sort of stuff that should live in `EventSerializer`. Maybe store the auth information in `request.auth`?